### PR TITLE
Ip helper

### PR DESCRIPTION
**Pull request by mpcarolin**
## Context

* I have been wanting a task for helping with ip addresses

## Updates
* `repos/cli-utils/src/network/getPrivateIPs.js`
  * gets all private ips for the `en0` interface
* `repos/cli-utils/src/network/getAddresses.js`
  * helper for filtering the results of `os.networkInterfaces()`
* `repos/cli-utils/src/network/getHostIP.js`
  * just renamed and moved, used to be `getInternalIp`
  * also updated it to use `mapFind` 
* `repos/cli-utils/src/network/getPublicIP.js`
  * gets the current machine's public ip
* `repos/cli-utils/src/network/getPublicIPsForUrl.js`
  * returns the public ip for a given url
* `repos/cli-utils/src/network/ipIsInRange.js`
  * helper checking if an ip address falls within a range
* `src/tasks/network/ip.js`
  * new ip task for getting data about ip addresses

## Testing

* verify unit tests pass
* `keg net ip`
  * verify it prints out your public and one private ip
* `keg net ip --private`
   * verify it prints out only the private ip
* `keg net ip --public`
   * verify it prints out only the public ip
* `keg net ip --url archive.org`
  * verify it prints out the public ip for archive.or

### CODE REVIEW

File: repos/cli-utils/src/network/isPrivateIP.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { ipIsInRange } = require('./ipIsInRange')
+
+const privateIpv4Classes = {
```
Comment (lancetipton) [inline repos/cli-utils/src/network/isPrivateIP.js:3]: This should go into the constants. 
* It's doubtful they will be reused, but it is possible, 
* Also, follows the pattern of other hard coded values

File: repos/cli-utils/src/network/getPublicIPsForUrl.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const dns = require('dns')
+
+/**
+ * @param {String} url 
+ * @returns {Promise} a promise that resolves to the ip address mapped to the url
+ */
+const getPublicIPsForUrl = url => {
```
Comment (lancetipton) [inline repos/cli-utils/src/network/getPublicIPsForUrl.js:7]: Is this actually being used anywhere?
* I'm not against it, I'm just curious for why it was added.

**Review by lancetipton — CHANGES_REQUESTED** [review]
Looks good, there's a couple small changes. Thanks

File: repos/cli-utils/src/network/__tests__/getAddresses.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+
+jest.mock('os')
+
+const { getAddresses } = require('../')
+const os = require('os')
+
+const privateAddress = {
```
[the remark this answers is not in the document]
Comment (mpcarolin) [reply repos/cli-utils/src/network/__tests__/getAddresses.js]: Thanks. Updated

File: repos/cli-utils/src/network/isPrivateIP.js (hunk 1 above)
Comment (mpcarolin) [reply repos/cli-utils/src/network/isPrivateIP.js]: Updated

**Review by lancetipton — APPROVED** [review]
Looks good, thanks

**Pull request merged**